### PR TITLE
Node E2E: Make readiness check handling process exits with 0 exit code.

### PR DESCRIPTION
**Pull request by Random-Liu**
As is mentioned by @mtaufen:
 "there is a problem with the way service `start` is currently implemented in test/e2e_node/e2e_service.go. If the Kubelet exits with status 0 before the health check completes, cmdErrorChan will be closed and, as a result, nil will be read from that channel, and you will return a nil error from `start`."

This PR changes the logic to:
1) If the err channel returns an error, return the error
2) If the err channel returns a nil, ignore it and continue checking readiness.
3) If the err channel is closed before readiness check succeeds, replace it with `blockCh` and continue checking readiness.

@mtaufen 
/cc @kubernetes/sig-node

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30913)

### CODE REVIEW

File: test/e2e_node/e2e_service.go [hunk 1]
```diff
@@ -454,10 +454,21 @@ func (s *server) String() string {
 // TODO(random-liu): Move this to util
 func readinessCheck(urls []string, errCh <-chan error) error {
 	endTime := time.Now().Add(*serverStartTimeout)
```
Comment (mtaufen) [inline test/e2e_node/e2e_service.go]: Add the systemd-related comment from #30915 and then looks good to me.

**Comment by Random-Liu** [comment]
@mtaufen Addressed comment. :) Thanks for reviewing!

**Comment by mtaufen** [comment]
LGTM

**Comment by mtaufen** [comment]
thanks!

**Comment by dchen1107** [comment]
Looks like your pr hits the integration flakiness at merging time. Please rerun the tests.

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #30947.

**Comment by Random-Liu** [comment]
@k8s-bot test this issue.

**Pull request merged**